eds/extend.py

```diff
@@ -132,5 +134,7 @@ def is_installed(pip_install_requirement: str) -> bool:
     Returns:
         bool: Whether a pip install requirement is installed.
     """
-    # todo: implement using https://github.com/pypa/pip/blob/main/src/pip/_internal/req/req_install.py#L392
-    raise NotImplementedError()
+    req = Requirement(pip_install_requirement)
+    install_req = InstallRequirement(req, None)
```

Comment:
I'd be inclined to use something more along the lines of feature detection here. There is a common pattern of wrapping an import statement in a `try...except ImportError...` block, which might look something like:

```
try:
    // using https://docs.python.org/3/library/importlib.html
    // import importlib at top of file
    importlib.import_module('os.path')
    return true
except ImportError as e:
    log.debug(e)
    return false

This would remove the dependency, not just on an internal class, but on Pip entirely.

Reply:
pip can be assumed, since its registered as an install requirement for EDS.

Reply:
While I cringed a bit at the ``from pip._internal`` and ``from pip._vendor`` imports, taking this approach as opposed to ``import_module`` does, as far as I know, avoid actually importing the requirement. That might be a good thing.